### PR TITLE
Deploy or upgrade facets with initialize arguments

### PR DESCRIPTION
**Pull request by zajck**
So far, deployment and upgrade script assumed that all facets have initalizers without arguments. The only exception was `configHandlerFacet` which has its own script `deploy-protocol-config-facet`.

This PR introduces new method `deployProtocolHandlerFacetsWithArgs` where you can pass in the initializer arguments.
Existing methods `deployProtocolHandlerFacets` and `deployProtocolConfigFacet` were refactored so now they both use `deployProtocolHandlerFacetsWithArgs`. They were still kept in, since they are used all over the code and replacing them everywhere would be tedious and error prone.

I had to slightly modify `scripts/config/facet-upgrade.js` and I added a new config file `scripts/config/facet-deploy.js`

In this branch only the configHandler accepts the arguments, so you can test only with it. In my other branch https://github.com/bosonprotocol/boson-protocol-contracts/tree/whitelist-metatx-initializer `MetaTransactionHandler` also accepts init args, and that branch has the same deploy/upgrade script as this, so you can test there as well.

### CODE REVIEW

**Review by anajuliabit — DISMISSED** [review]
I ran the scripts in your other branch and on this one as well. Everything works fine.

LGTM

File: docs/local-development.md [hunk 1]
```diff
@@ -100,7 +100,9 @@ To perform the upgrade you then
 - Update config file `scripts/config/facet-upgrade.js`:
   - "addOrUpgrade" is the list of facets that will be upgraded or added,
   - "remove": list of facets that will be completely removed
-  - "skip" allows you to specify methods that will be ignored during the process.
+  - "skipSelectors" allows you to specify methods that will be ignored during the process.
+  - "initArgs": if facet initializer expects arguments, provide it here. For no-arg initializers you don't have to specify anything.
```
Comment (hswopeams) [inline docs/local-development.md:104]:
```suggestion
  - "initArgs": if facet initializer expects arguments, provide them here. For no-arg initializers you don't have to specify anything.
```

File: scripts/config/facet-deploy.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/**
+ * Config file used to upgrade the facets
```
Comment (hswopeams) [inline scripts/config/facet-deploy.js:2]:
```suggestion
 * Config file used to deploy the facets
```

File: scripts/config/facet-upgrade.js [hunk 3]
```diff
@@ -3,12 +3,16 @@
  *
  * - addOrUpgrade: list of facets that will be upgraded or added
  * - remove: list of facets that will be completely removed
- * - skip:  mapping "facetName":"listOfFunctionsToBeSkipped". With this you can specify functions that will be ignored during the update.
+ * - skipSelectors:  mapping "facetName":"listOfFunctionsToBeSkipped". With this you can specify functions that will be ignored during the update.
  *          You don't have to specify "initialize()" since it's ignored by default.
  *          Skip does not apply to facets that are completely removed.
+ * - initArgs: if facet initializer expects arguments, provide it here. For no-arg initializers you don't have to specify anything.
```
Comment (hswopeams) [inline scripts/config/facet-upgrade.js:9]:
```suggestion
 * - initArgs: if facet initializer expects arguments, provide them here. For no-arg initializers you don't have to specify anything.
```

File: scripts/config/facet-upgrade.js [hunk 4]
```diff
@@ -3,12 +3,16 @@
...
+ * - skipInit": list of facets for which you want to skip initialization call.
  */
 exports.Facets = {
-  addOrUpgrade: ["SellerHandlerFacet", "BuyerHandlerFacet"],
+  addOrUpgrade: [],
```
Comment (hswopeams) [inline scripts/config/facet-upgrade.js:13]: I think it makes sense to remove the examples from the actual code, but I think it's a good idea to have an example config in the comments. This is for all of the options.

File: scripts/config/facet-deploy.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    "ExchangeHandlerFacet",
+    "FundsHandlerFacet",
+    "GroupHandlerFacet",
+    "OfferHandlerFacet",
+    "OrchestrationHandlerFacet",
+    "TwinHandlerFacet",
+    "PauseHandlerFacet",
+    "MetaTransactionsHandlerFacet",
+  ],
+  argFacets: {},
```
Comment (hswopeams) [inline scripts/config/facet-deploy.js:25]: I think it's a good idea to have an example config in the comments.

**Review by hswopeams — CHANGES_REQUESTED** [review]
Looks good. I have a few requests regarding the comments.

File: docs/local-development.md (hunk 1 above)
Comment (zajck) [reply docs/local-development.md]: Done.

File: scripts/config/facet-deploy.js (hunk 2 above)
Comment (zajck) [reply scripts/config/facet-deploy.js]: Done.

File: scripts/config/facet-deploy.js (hunk 5 above)
Comment (zajck) [reply scripts/config/facet-deploy.js:34]: Done.

File: scripts/config/facet-upgrade.js (hunk 3 above)
Comment (zajck) [reply scripts/config/facet-upgrade.js]: Done

File: scripts/config/facet-upgrade.js (hunk 4 above)
Comment (zajck) [reply scripts/config/facet-upgrade.js:25]: Done.

**Review by hswopeams — APPROVED** [review]

**Review by anajuliabit — APPROVED** [review]

**Pull request merged**